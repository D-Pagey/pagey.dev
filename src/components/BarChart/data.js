export default [
    { id: '1', value: 58 },
    { id: '2', value: 67 },
    { id: '3', value: 58 },
    { id: '4', value: 63 },
    { id: '5', value: 63 },
    { id: '6', value: 62 },
    { id: '7', value: 52 },
    { id: '8', value: 54 }
];
