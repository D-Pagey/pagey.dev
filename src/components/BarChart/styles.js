import styled from 'styled-components/macro';
import { animated } from 'react-spring';

export const SVG = styled.svg`
    max-width: 500px;
`;

export const Bar = styled(animated.rect)`
    fill: skyblue;
    transition: 0.3s;

    &:hover {
        fill: navy;
    }
`;
