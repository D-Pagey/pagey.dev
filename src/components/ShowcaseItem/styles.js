import styled from 'styled-components/macro';

export const Wrapper = styled.div`
    border-top: 1px solid #d2d2d2;
    display: grid;
    grid-row-gap: 1rem;
    grid-template-columns: 1fr;
    padding: 1rem 0;

    @media (min-width: 700px) {
        grid-template-columns: 1fr 400px;
        grid-template-rows: max-content;
        padding: 1rem 0.5rem;
    }
`;

export const Title = styled.h4`
    margin: 0;
`;

export const Description = styled.p`
    margin: 0;
`;

export const Thumbnail = styled.div`
    grid-row: 1;
    max-width: 400px;

    @media (min-width: 700px) {
        grid-column: 2;
        grid-row: 1 / 3;
    }
`;
