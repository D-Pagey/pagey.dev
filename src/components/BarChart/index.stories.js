import React from 'react';
import { storiesOf } from '@storybook/react';
import BarChart from '.';

const props = {
    height: 400,
    paddingX: 50,
    paddingY: 35,
    width: 500
};

storiesOf('BarChart', module).add('default', () => <BarChart {...props} />);
