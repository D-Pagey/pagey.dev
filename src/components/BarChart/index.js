import React from 'react';
import { scaleLinear } from 'd3-scale';
import * as S from './styles';

const data = [58, 67, 58, 63, 63, 62, 52];
const dataAvg = data.reduce((a, b) => a + b, 0) / data.length;

const width = 500;
const height = 400;
const paddingX = 50;
const paddingY = 35;

const BarChart = () => {
    const xScale = scaleLinear()
        .domain([0, data.length - 1])
        .range([0 + paddingX, width - 25]);

    const yScale = scaleLinear()
        .domain([Math.min(...data) - 2, Math.max(...data)])
        .range([0, height - paddingY - paddingY]);

    return (
        <div>
            <h1>My Gym Usage:</h1>
            <svg width={width} height={height}>
                <line
                    x1={paddingX}
                    y1={yScale(dataAvg)}
                    x2={width}
                    y2={yScale(dataAvg)}
                    stroke="red"
                    strokeWidth="1"
                    strokeDasharray="2"
                />
                <line x1={paddingX} y1={height - paddingY} x2={paddingX} y2={paddingY} stroke="grey" strokeWidth="1" />
                <text x={paddingX - 5} y={paddingY + 5} textAnchor="end">
                    {Math.max(...data)}
                </text>
                <text x={paddingX - 50} y={height / 2}>
                    Minutes
                </text>

                <text x={paddingX - 5} y={height - paddingY + 5} textAnchor="end">
                    {Math.min(...data) - 2}
                </text>
                <text x={width / 2} y={height}>
                    Index
                </text>

                <line
                    x1={paddingX}
                    y1={height - paddingY}
                    x2={width}
                    y2={height - paddingY}
                    stroke="grey"
                    strokeWidth="1"
                />
                <text x={width - 60} y={height / 2}>
                    Average
                </text>

                {data.map((item, index) => (
                    // eslint-disable-next-line react/no-array-index-key
                    <g key={index}>
                        <S.Bar
                            height={yScale(item)}
                            x={xScale(index)}
                            y={height - yScale(item) - paddingY}
                            width={30}
                        />
                        <text x={xScale(index) + 8} y={height - paddingY + 15}>
                            {index}
                        </text>
                    </g>
                ))}
            </svg>
        </div>
    );
};

export default BarChart;
