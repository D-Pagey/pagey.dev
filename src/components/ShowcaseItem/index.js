import React from 'react';
import * as S from './styles';

const ShowcaseItem = () => (
    <S.Wrapper>
        <S.Title>Bar Chart</S.Title>
        <S.Copy>This is a bar chart showing my gym usage.</S.Copy>

        <S.Placeholder />
    </S.Wrapper>
);

export default ShowcaseItem;
