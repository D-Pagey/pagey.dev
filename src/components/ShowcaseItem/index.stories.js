import React from 'react';
import { storiesOf } from '@storybook/react';
import BarChart from '../BarChart';
import ShowcaseItem from '.';

const props = {
    description: ['A description of the showcase', 'Another paragraph'],
    thumbnail: <BarChart height={400} paddingX={50} paddingY={35} width={500} />,
    title: 'Showcase Title'
};

storiesOf('ShowcaseItem', module).add('default', () => <ShowcaseItem {...props} />);
