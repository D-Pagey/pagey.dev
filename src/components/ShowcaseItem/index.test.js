import React from 'react';
import BarChart from '../BarChart';
import ShowcaseItem from '.';

const props = {
    description: ['A description of the showcase', 'Another paragraph'],
    thumbnail: <BarChart height={400} paddingX={50} paddingY={35} width={500} />,
    title: 'Showcase Title'
};

describe('ShowcaseItem component', () => {
    it('should render', () => {
        const { container } = render(<ShowcaseItem {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
