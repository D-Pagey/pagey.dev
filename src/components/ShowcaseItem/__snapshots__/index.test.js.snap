// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowcaseItem component should render 1`] = `
.c4 {
  max-width: 500px;
}

.c5 {
  fill: skyblue;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c5:hover {
  fill: navy;
}

.c0 {
  border-top: 1px solid #f3f3f3;
  display: grid;
  grid-row-gap: 1rem;
  grid-template-columns: 1fr;
  padding: 1rem 0;
}

.c1 {
  margin: 2rem 0 0;
}

.c2 {
  margin: 0 0 1rem;
}

.c3 {
  grid-row: 1;
  max-width: 400px;
}

@media (min-width:700px) {
  .c0 {
    grid-column-gap: 1rem;
    grid-template-columns: 1fr 400px;
    grid-template-rows: max-content;
    padding: 1rem 0.5rem;
  }
}

@media (min-width:700px) {
  .c3 {
    grid-column: 2;
    grid-row: 1 / 3;
  }
}

<div
  class="c0"
>
  <h4
    class="c1"
  >
    Showcase Title
  </h4>
  <div>
    <p
      class="c2"
    >
      A description of the showcase
    </p>
    <p
      class="c2"
    >
      Another paragraph
    </p>
  </div>
  <div
    class="c3"
  >
    <svg
      class="c4"
      viewBox="0 0 500 400"
    >
      <text
        x="420"
        y="176.66666666666666"
      >
        Average
      </text>
      <line
        stroke="green"
        stroke-dasharray="2"
        stroke-width="1"
        x1="50"
        x2="450"
        y1="181.66666666666666"
        y2="181.66666666666666"
      />
      <line
        stroke="grey"
        stroke-width="1"
        x1="50"
        x2="50"
        y1="35"
        y2="365"
      />
      <text
        text-anchor="end"
        x="45"
        y="40"
      >
        68
      </text>
      <text
        text-anchor="end"
        x="45"
        y="186.66666666666666"
      >
        60
      </text>
      <text
        transform="rotate(270) translate(-225, -180)"
        x="0"
        y="200"
      >
        Minutes
      </text>
      <text
        text-anchor="end"
        x="45"
        y="370"
      >
        50
      </text>
      <text
        x="250"
        y="398"
      >
        Index
      </text>
      <line
        stroke="grey"
        stroke-width="1"
        x1="50"
        x2="480"
        y1="365"
        y2="365"
      />
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="146.66666666666666"
          rx="3"
          width="30"
          x="50"
          y="218.33333333333334"
        />
        <text
          x="58"
          y="380"
        >
          0
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="311.6666666666667"
          rx="3"
          width="30"
          x="107.14285714285714"
          y="53.333333333333314"
        />
        <text
          x="115.14285714285714"
          y="380"
        >
          1
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="146.66666666666666"
          rx="3"
          width="30"
          x="164.28571428571428"
          y="218.33333333333334"
        />
        <text
          x="172.28571428571428"
          y="380"
        >
          2
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="238.33333333333334"
          rx="3"
          width="30"
          x="221.42857142857142"
          y="126.66666666666666"
        />
        <text
          x="229.42857142857142"
          y="380"
        >
          3
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="238.33333333333334"
          rx="3"
          width="30"
          x="278.57142857142856"
          y="126.66666666666666"
        />
        <text
          x="286.57142857142856"
          y="380"
        >
          4
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="220"
          rx="3"
          width="30"
          x="335.7142857142857"
          y="145"
        />
        <text
          x="343.7142857142857"
          y="380"
        >
          5
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="36.666666666666686"
          rx="3"
          width="30"
          x="392.85714285714283"
          y="328.3333333333333"
        />
        <text
          x="400.85714285714283"
          y="380"
        >
          6
        </text>
      </g>
      <g>
        <rect
          class="c5"
          data-testid="barChart"
          height="73.33333333333331"
          rx="3"
          width="30"
          x="450"
          y="291.6666666666667"
        />
        <text
          x="458"
          y="380"
        >
          7
        </text>
      </g>
    </svg>
  </div>
</div>
`;
